include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -588,22 +706,33 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<Swizzle,Dither,Nor
 		)
 		{
 			// TODO: investigate properly if its supposed be `size` or `size-1` (polyphase kinda shows need for `size`)
-			core::vectorSIMDi32 last(kernelX.getWindowSize().x,0,0,0);
+			core::vectorSIMDi32 last(kernelX.getWindowSize().x-1,0,0,0);
 			if (inImageType>=IImage::ET_2D)
-				last.y = kernelY.getWindowSize().x;
+				last.y = kernelY.getWindowSize().y-1;
 			if (inImageType>=IImage::ET_3D)
-				last.z = kernelZ.getWindowSize().x;
+				last.z = kernelZ.getWindowSize().z-1;
 			return last;
 		}
+		static inline core::vectorSIMDu32 getPhaseSupportLUTAxisOffsets(const core::vectorSIMDu32& phaseCount,
+			const CScaledImageFilterKernel<KernelX>& kernelX,
+			const CScaledImageFilterKernel<KernelY>& kernelY,
+			const CScaledImageFilterKernel<KernelZ>& kernelZ)
+		{
+			core::vectorSIMDu32 result;
+			result.x = 0u;
+			result.y = (phaseCount[0] * kernelX.getWindowSize().x) * sizeof(value_type);
+			result.z = ((phaseCount[0] * kernelX.getWindowSize().x) + (phaseCount[1] * kernelY.getWindowSize().y)) * sizeof(value_type);
+			return result;
+		}
 		// the blit filter will filter one axis at a time, hence necessitating "ping ponging" between two scratch buffers
-		static inline uint32_t getScratchOffset(const state_type* state, bool secondPong)
+		static inline uint32_t getScratchOffset(const state_type* state, bool secondPong,
+			const CScaledImageFilterKernel<KernelX>& scaledKernelX,
+			const CScaledImageFilterKernel<KernelY>& scaledKernelY,
+			const CScaledImageFilterKernel<KernelZ>& scaledKernelZ)
 		{
 			const auto inType = state->inImage->getCreationParameters().type;
-			const auto kernelX = state->contructScaledKernel(state->kernelX);
-			const auto kernelY = state->contructScaledKernel(state->kernelY);
-			const auto kernelZ = state->contructScaledKernel(state->kernelZ);
 
-			const auto window_end = getWindowEnd(state->inImage->getCreationParameters().type,kernelX,kernelY,kernelZ);
+			const auto window_end = getWindowEnd(state->inImage->getCreationParameters().type,scaledKernelX,scaledKernelY,scaledKernelZ);
 			// TODO: account for the size needed for coverage adjustment
 			// the first pass will be along X, so new temporary image will have the width of the output extent, but the height and depth will need to be padded
 			// but the last pass will be along Z and the new temporary image will have the exact dimensions of `outExtent` which is why there is a `core::max`
```

Comment:
probably in the interest of dryness we should make a function called `computeIntermediateExtents`

because there's some duplication here in working out the intermediate extents and in the `execute()` call

Reply:
Resolved in #343.